front/src/modules/activities/tasks/components/TaskGroups.tsx

```diff
@@ -69,6 +72,14 @@ export function TaskGroups({ entity, showAddButton }: OwnProps) {
     TasksRecoilScopeContext,
   );
 
+  const [, setDueTasks] = useRecoilState(dueTasksState);
+
+  useEffect(() => {
```

Comment:
This isn't working because we need to be on the Task page for notifications to appear.

You should rely on a query that will tap into Apollo's store or request tasks, wherever you are on the app. 

Maybe use useTask directly into a new component TaskNavMenuItem that returns a NavMenuItem but that calls useTasks to compute notification count. That way AppNavBar won't be bloated by a lot of hooks to compute notifications for different NavItems.